src/external/wannierio.jl

```diff
@@ -0,0 +1,693 @@
+using Dates
+using Printf: @sprintf
+using SpecialFunctions: erfc
+
+"""
+Return the kpoint index shift for spin-polarized case.
+
+For spin-polarized calculation, first half of kpoints are for spin up,
+and the second half are for spin down.
+"""
+function _get_ik_shift(basis::PlaneWaveBasis, spin::Int)
+    ik_shift = 0
+    n_spin = basis.model.n_spin_components
+    n_kpts = length(basis.kpoints) ÷ n_spin
+    # only for spin down case, the kpoint index needs to be shifted
+    spin == 2 && (ik_shift = n_kpts)
+    ik_shift
+end
+
+@doc raw"""
+Computes the overlap matrix for Wannierization.
+
+``[M^{k,b}]_{m,n} = \langle u_{m,k} | u_{n,k+b} \rangle``
+for given kpoint ``k`` and ``b`` vector.
+
+The ``k+b`` vector is actually shifted to its periodic image, denoted by
+another kpoint ``k`` inside the uniform kgrid, together with a shifting
+vector ``G_shift``.
+`G_shift` is the "shifting" vector, correction due to the periodicity conditions
+imposed on ``k \to  ψ_k``.
+It is non zero if ``k+b`` is taken in another unit cell of the reciprocal lattice.
+We use here that:
+``u_{n(k + G_{\rm shift})}(r) = e^{-i G_{\rm shift} \cdot r} u_{nk}``.
+
+# Arguments
+- `basis`: the `PlaneWaveBasis`
+- `ψ`: the Bloch states, a vector of length `n_kpts`, each element is
+    a matrixof size `(n_G, n_bands)`. In the spin-polarized case, the
+    vector is of length `2*n_kpts`, and the first half is for spin-up,
+    the second half is for spin-down.
+- `kpb_k`: the kpoint indices of k+b, size: `(n_bvecs, n_kpts)`.
+    1st index: b-vector, 2nd index: kpoint, e.g., `kpb_k[ib, ik]` is the
+    corresponding kpoint index of k+b for `ib`-th b-vector at `ik`-th kpoint.
+- `kpb_G`: the shifting vectors of k+b, size: `(3, n_bvecs, n_kpts)`.
+    1st index: x,y,z component of b-vector,
+    2nd index: b-vector,
+    3rd index: shifting vector to get the shifted k-point `kpb_k[ib,ik]` to
+    the actual k+b point, thus the actual k+b is `kpb_k[ib,ik] + kpb_G[:,ib,ik]`.
+
+# Keyword arguments
+- `spin`: the spin channel to be used, either `1` or `2`.
+"""
+@timing function compute_mmn(
```

Comment:
If this computes the overlaps, than that's what it should be called. Julia allows more than 8 character function names ;).